tests/unit/test_charm.py

```diff
@@ -12,9 +12,109 @@ def setUp(self) -> None:
         self.harness.begin()
         self.maxDiff = None
 
+    @patch("ops.framework.EventBase.defer")
+    def test_config_available_fail(self, defer):
+        self.harness.charm._slurmrestd.on.config_available.emit()
+        defer.assert_called()
+
+    @patch("slurm_ops_manager.SlurmManager.render_slurm_configs")
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_slurm_config", return_value={"cluster_name": "test"})
+    @patch("slurm_ops_manager.SlurmManager.needs_reboot", new_callable=PropertyMock(return_value=False))
+    @patch("interface_slurmrestd.SlurmrestdRequires.is_joined", new_callable=PropertyMock(return_value=True))
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_munge_key", lambda _: True)
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_jwt_rsa", lambda _: True)
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_slurm_config", lambda _: True)
+    @patch("ops.framework.EventBase.defer")
+    def test_config_available_success(self, defer, *_):
+        self.harness.charm._stored.slurm_installed = True
+        self.harness.charm._stored.slurmrestd_restarted = True
+
+        self.harness.charm._slurmrestd.on.config_available.emit()
+        defer.assert_not_called()
+
+    def test_config_unavailable(self):
+        self.harness.charm._slurmrestd.on.config_unavailable.emit()
+        self.assertFalse(self.harness.charm._stored.slurmrestd_restarted)
+
+    @patch("pathlib.Path.read_text")
+    @patch("ops.model.Unit.set_workload_version")
+    @patch("ops.framework.EventBase.defer")
+    def test_install_fail(self, defer, *_):
+        self.harness.charm.on.install.emit()
+        self.assertFalse(self.harness.charm._stored.slurm_installed)
+        defer.assert_called()
+
+    @patch("slurm_ops_manager.SlurmManager.needs_reboot", new_callable=PropertyMock(return_value=False))
+    @patch("interface_slurmrestd.SlurmrestdRequires.is_joined", new_callable=PropertyMock(return_value=True))
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_munge_key", lambda _: True)
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_jwt_rsa", lambda _: True)
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_slurm_config", lambda _: True)
+    @patch("slurm_ops_manager.SlurmManager.install")
+    @patch("pathlib.Path.read_text")
+    @patch("ops.model.Unit.set_workload_version")
+    @patch("slurm_ops_manager.SlurmManager.start_munged", lambda _: True)
+    @patch("ops.framework.EventBase.defer")
+    def test_install_success(self, defer, *_):
+        self.harness.charm.on.install.emit()
+        self.assertTrue(self.harness.charm._stored.slurm_installed)
+        self.assertEqual(self.harness.charm.unit.status, ActiveStatus("slurmrestd available"))
+        defer.assert_not_called()
```

Comment:
Similar to the `slurmd` charm, do we really want to mock installing `slurmrestd`. This method success is hinged on `SlurmManager.install` working for `slurmrestd`. I think you should make sure that the install method at least works.

Also, not sure if we want to be patching out that the version file is missing. I think how version is set at install is a major functional issue that needs to be address when we start refactoring the slurm charms.

Reply:
No longer patching out the version file